thirdeye-ui/src/app/components/alert-wizard-v2/alert-notifications/subscription-groups/subscription-groups.component.tsx

```diff
@@ -51,7 +51,12 @@ export const SubscriptionGroups: FunctionComponent<SubscriptionGroupsProps> = ({
         getSubscriptionGroups().then((fetchedSubscriptionGroups) => {
             const subscribed = fetchedSubscriptionGroups
                 ? fetchedSubscriptionGroups.filter((group) => {
-                      return group.alerts.some((item) => item.id === alert.id);
+                      const alerts =
+                          group.alertAssociations?.map((a) => a.alert) ||
+                          group.alerts ||
+                          [];
+
+                      return alerts.some((item) => item.id === alert.id);
                   })
```

Comment:
For sake of readability at a glance, I would change this to:

```
let subscribed = [];

if (fetchedSubscriptionGroups) {
    subscribed = fetchedSubscriptionGroups.filter((group) => {
        return group.alerts.some((item) => item.id === alert.id);
        const alerts =
            group.alertAssociations?.map((a) => a.alert) ||
            group.alerts ||
            [];

        return alerts.some((item) => item.id === alert.id);
    })
}
```

Reply:
Updated accordingly